python/rikai/spark/sql/callback_service.py

```diff
@@ -0,0 +1,82 @@
+#  Copyright 2021 Rikai Authors
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from py4j.java_gateway import CallbackServerParameters
+from pyspark.sql import SparkSession
+
+from rikai.logging import logger
+
+__all__ = ["init_cb_service"]
+
+
+def init_cb_service(spark: SparkSession):
+    jvm = spark.sparkContext._gateway
+    params = CallbackServerParameters(
+        daemonize=True,
+        daemonize_connections=True,
+        auth_token=jvm.gateway_parameters.auth_token,
+    )
+    jvm.start_callback_server(callback_server_parameters=params)
+    logger.info("Spark callback server started")
+
+    cb = CallbackService(spark)
+    cb.register()
+    logger.info("Rikai Python callback service registered")
+
+
+class CallbackService(object):
```

Comment:
nit: i don't think it's required anymore to inherit from `object`

Reply:
fixed

Reply:
fixed